pkg/domain/domain.go

```diff
@@ -2058,7 +2058,12 @@ func (do *Domain) LoadPrivilegeLoop(sctx sessionctx.Context) error {
 	return nil
 }
 
-func privReloadEvent(h *privileges.Handle, event *PrivilegeEvent) (err error) {
+func (do *Domain) privReloadEvent(h *privileges.Handle, event *PrivilegeEvent) (err error) {
+	if event.ServerID == do.ServerID() {
```

Comment:
We need to make do.ServerID() never be 0, otherwise it would be buggy here.

Reply:
Done